stacks/minstack.js

```diff
@@ -0,0 +1,39 @@
+const Stack = require('./Stack');
+
+function MinStack(stackMaxSize){
+    this.stack = new Stack(stackMaxSize);
+    this.minStack = new Stack(stackMaxSize);
+}
+
+MinStack.prototype.push = function(num) {
+    var currentMinimum = this.minStack.peek();
+    if(this.minStack.count() === 0 || (!currentMinimum || currentMinimum > num) ) //First element
+    {
+        this.minStack.push(num);
+    }
+    else
+    {
+        this.minStack.push(currentMinimum);
+    }
+    this.stack.push(num);
+}
+
+MinStack.prototype.pop = function() {
+    this.minStack.pop();
+    this.stack.pop();
+}
+
+MinStack.prototype.min = function() {
+    if( this.minStack.count() === 0 )
```

Comment:
dont add console logs, this would not work with your tests anyways.

Reply:
removed